kani-compiler/src/kani_queries/mod.rs

```diff
@@ -41,4 +44,40 @@ impl QueryDb {
     pub fn args(&self) -> &Arguments {
         self.args.as_ref().expect("Arguments have not been initialized")
     }
+
+    pub fn add_modifies_contract(&mut self, name: InternedString, contract: AssignsContract) {
+        assert!(
+            self.modifies_contracts.insert(name, contract).is_none(),
+            "Invariant broken, tried adding second modifies contracts to: {name}",
+        )
+    }
+
+    pub fn get_modifies_contracts(&mut self, key: &str) -> Option<AssignsContract> {
+        self.modifies_contracts.remove(&key.intern())
+    }
+
+    pub fn assert_modifies_contracts_received(&self) {
```

Comment:
Why do we need this? I don't understand why users of this class should be invoking this function.

Reply:
The naming might not be ideal. This works in conjunction with the fact that `get_modifies_contract` removes the contract to ensure that all contracts were handled. 

I've added documentation explaining this

Reply:
Can you retrieve the map of contracts instead inside `kani-compiler`? Removing an item as part of the `get` method is fragile and the interface of this object is trying to reflect the one specific use case you have right now.